legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/depot/DepotConfiguration.java

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.server.depot;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.finos.legend.sdlc.server.depot.auth.AuthClientInjector;
+
+public class DepotConfiguration
+{
+    private final ServerConfiguration serverConfiguration;
+    private final AuthClientInjector authClientInjector;
+
+    private DepotConfiguration(ServerConfiguration serverConfiguration, AuthClientInjector authClientInjector)
+    {
+        this.serverConfiguration = serverConfiguration;
+        this.authClientInjector = authClientInjector;
+    }
+
+    public ServerConfiguration getServerConfiguration()
+    {
+        return this.serverConfiguration;
+    }
+
+    public AuthClientInjector getAuthClientInjector()
+    {
+        return this.authClientInjector;
+    }
+
+    public static DepotConfiguration emptyConfiguration()
+    {
+        return new DepotConfiguration(ServerConfiguration.emptyConfiguration(), (HttpClientBuilder builder) -> builder);
```

Comment:
It's probably better to just use null for the auth client injector here. You should probably resolve it to a default value at a later point.

Reply:
moved setting default value to AbstractBaseModule, is that okay or should I resolve it even later?